src/python/bot/fuzzers/syzkaller/fuzzer.py

```diff
@@ -0,0 +1,167 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""syzkaller fuzzer."""
+from __future__ import absolute_import
+from base import utils
+from bot.fuzzers import builtin
+from bot.fuzzers import utils as fuzzer_utils
+from bot.fuzzers.syzkaller import constants
+from bot.fuzzers.syzkaller import generate_config
+from builtins import object
+from system import environment
+from system import new_process
+import copy
+import re
+import tempfile
+
+# Regex to find testcase path from a crash.
+KASAN_CRASH_TESTCASE_REGEX = (r'.*Test unit written to\s*'
+                              r'(Read|Write) of .*')
+
+
+def get_arguments(fuzzer_path):
+  """Get arguments for a given fuzz target."""
+  del fuzzer_path
+  build_dir = environment.get_value('BUILD_DIR')
+  device_serial = environment.get_value('ANDROID_SERIAL')
+  json_config_path = '/tmp/' + device_serial + '/config.json'
+  generate_config.run(
+      serial=device_serial,
+      work_dir_path='/tmp/syzkaller',
+      binary_path=build_dir + '/syzkaller',
+      vmlinux_path='/tmp/syzkaller/vmlinux',
+      config_path=json_config_path,
+      kcov=True,
+      reproduce=False)
+  arguments = ['--config', json_config_path]
+  return arguments
+
+
+def get_runner(fuzzer_path):
+  """Return a suzkaller runner object."""
+  build_dir = environment.get_value('BUILD_DIR')
+  return SyzkallerRunner(fuzzer_path, build_dir)
+
+
+class Syzkaller(builtin.EngineFuzzer):
+  """Builtin syzkaller fuzzing engine."""
+
+  def generate_arguments(self, fuzzer_path):
+    """Generate arguments for fuzzer using .options file or default values."""
+    return ' '.join(get_arguments(fuzzer_path))
+
+
+class SyzkallerRunner(new_process.ProcessRunner):
+  """Syzkaller runner."""
+
+  def __init__(self, executable_path, default_args=None):
+    """Inits the SyzkallerRunner.
+
+    Args:
+      executable_path: Path to the fuzzer executable.
+      default_args: Default arguments to always pass to the fuzzer.
+    """
+    super(SyzkallerRunner, self).__init__(
+        executable_path=executable_path, default_args=None)
+
+  def get_command(self, additional_args=None):
+    """Process.get_command override."""
+    base_command = super(SyzkallerRunner,
+                         self).get_command(additional_args=additional_args)
+
+    return base_command
+
+  def _create_empty_testcase_file(self):
+    """Create an empty testcase file in temporary directory."""
+    _, path = tempfile.mkstemp(dir=fuzzer_utils.get_temp_dir())
+    return path
+
+  def get_testcase_path(self, log_lines):
+    """Get testcase path from log lines."""
+    for line in log_lines:
+      match = re.match(KASAN_CRASH_TESTCASE_REGEX, line)
+      if match:
+        return match.group(1)
+
+    return None
+
+  def fix_timeout_argument_for_reproduction(self, arguments):
```

Comment:
this is copied from libFuzzer, but is it really necessary here? Also, TIMEOUT_FLAG is not defined in constants.py?

Reply:
Removed and will be added once the repro logic is added.